splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceUnitTestWithTempDirectory.java

```diff
@@ -0,0 +1,72 @@
+package com.splicemachine.derby.test.framework;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/// this class extends SpliceUnitTest so that each test
+/// can access a temporary output directory that is removed after the test is run
+/// the temporary output directory is created only on demand.
+/// might be also doable like SpliceSchemaWatcher extends TestWatcher
+public class SpliceUnitTestWithTempDirectory extends SpliceUnitTest
+{
+    /// set the following boolean to true to prevent deletion of temporary files (e.g. for debugging)
+    public boolean debug_no_delete = false;
+    /// the temporary directory (created on demand)
+    private Path tempDir = null;
+
+    /// this will be called after each test, deleting possible temp directories.
+    @After
+    public void deleteTempDirectory() throws Exception
+    {
+        if( tempDir == null )
+            return;
+        else if( debug_no_delete )
+            System.out.println( "Not deleting test temporary directory " + tempDir );
+        else
+            FileUtils.deleteDirectory( tempDir.toFile() );
+    }
+
+    /// this will return the temp directory, that is created on demand once for each test
+    public String getTempOutputDirectory() throws Exception
+    {
+        if( tempDir == null)
+            tempDir = Files.createTempDirectory( "SpliceUnitTest_temp_" );
```

Comment:
Better create temp directories in "/target" as previously - build is self-contained, more control, less surprises.

Reply:
ok i didn't do much java/maven before, for me it was a bit surprising that target is used for these kind of temporary directories, test output, current database state, logs, and not only as a target for binaries. actually a reason we have to advice to do mvn **clean** every time, to clean off the target dir - if i understood correctly.
But also works for me. With the current solution target dir should be cleaned after a test ran.

Reply:
done